src/helper/ssg/index.ts

```diff
@@ -67,17 +67,21 @@ interface SSGParam {
   [key: string]: string
 }
 type SSGParams = SSGParam[]
+
 interface SSGParamsMiddleware {
-  (generateParams: (c: Context) => SSGParams | Promise<SSGParams>): MiddlewareHandler
+  <E extends Env = Env>(
+    generateParams: (c: Context<E>) => SSGParams | Promise<SSGParams>
+  ): MiddlewareHandler<E>
 }
+
 type AddedSSGDataRequest = Request & {
   ssgParams?: SSGParams
 }
 /**
  * Define SSG Route
  */
-export const ssgParams: SSGParamsMiddleware = (generateParams) => async (c, next) => {
-  ;(c.req.raw as AddedSSGDataRequest).ssgParams = await generateParams(c)
+export const ssgParams: SSGParamsMiddleware = (params) => async (c, next) => {
+  ;(c.req.raw as AddedSSGDataRequest).ssgParams = Array.isArray(params) ? params : await params(c)
```

Comment:
Should you rewrite these lines? I think `generateParams` is always `function`.

Reply:
Understood. If you want to pass an array, write the overload signature to the interface declaration!